### ISSUE TITLE
Disgaea 5 Complete (803600)

### ISSUE THREAD

**Issue by ghost** [ValveSoftware/Proton#1801]
# Compatibility Report
- Name of the game with compatibility issues: Disgaea 5 Complete
- Steam AppID of the game:803600

## System Information
- GPU: GTX 1080
- Driver/LLVM version: NVIDIA 410.66
- Kernel version: 4.18.16-1
- Link to full system information report as [Gist](https://gist.github.com/):https://gist.github.com/Inotamira/7d39b0798b8b6ee9f4189ed3da69adf8
- Proton version:3.16-3

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-803600.log](https://github.com/ValveSoftware/Proton/files/2502454/steam-803600.log)

## Symptoms 
The game either get's stuck on the opening screen with nothing but the cursor displayed, or, if Direct X 11 is used via proton command, the game will crash instantly and give the error shown in the pastebin here https://pastebin.com/fkAAzgJC . It seems to be an issue with the way the mfplat is implemented into both WINE and Proton.

## Reproduction
Run the game without any special commands, or with DX11 forced via proton variable.

**Comment by cybik** [comment]
Looks like a MediaFoundation issue, again :/

**Comment by ghost** [comment]
> Looks like a MediaFoundation issue, again :/

Yep, and I have NO idea how to solve it, if it's something that can even be solved without a complete rework of WINE

**Comment by cybik** [comment]
First they should actually get a base implementation of Windows Media Foundation, to be honest. That'd unlock a couple of new games, like #1240 // FighterZ

**Comment by Sbte** [comment]
I had a look at this bug since I wanted to play the game and it's usually quite easy to fix unimplemented function errors. But I got a completely different error, in both wine and proton:

https://pastebin.com/Q3wJJX29

I assumed that with "if Direct X 11 is used via proton command" you meant you just used `PROTON_USE_WINED3D=1`. However, this is not quite true. Instead of doing this (which gives the above error) you used wine-staging, which apparently fixes the bug I got, and gets a little further. At that point you run into the missing mfplat function. I'll now have a look if I can fix that.

It might be, however, that the screen will still be black due to missing d3d11 functionality (I get a huge amount of d3d11 fixmes). So it might be worthwhile to report the initial error (in DXVK) in the mean time at https://github.com/doitsujin/dxvk/issues

**Comment by Sbte** [comment]
Ok, I got around all of the mfplat issues at startup (it now just skips the video it is trying to play), and now have music playing in the background. I still have a black screen, however, but many Windows users seem to be experiencing this as well. Hopefully the black screen issue is the same as for the Windows users. Or else maybe DXVK can solve this.

I'll try to submit some mfplat patches to Wine.

@cybik I think you have no idea how much work that is. I needed about 2000 lines for the few functions this game needs. Also, Wine has a policy of not implementing methods if they are not used by any application that they know of.

**Comment by ghost** [comment]
> Ok, I got around all of the mfplat issues at startup (it now just skips the video it is trying to play), and now have music playing in the background. I still have a black screen, however, but many Windows users seem to be experiencing this as well. Hopefully the black screen issue is the same as for the Windows users. Or else maybe DXVK can solve this.
> 
> I'll try to submit some mfplat patches to Wine.
> 
> @cybik I think you have no idea how much work that is. I needed about 2000 lines for the few functions this game needs. Also, Wine has a policy of not implementing methods if they are not used by any application that they know of.

No, I keep trying to find the command I found, but there's a specific set of commands specifically for DXVK within proton that force the mode that's used. The command in question forced DX11 mode for DXVK. It was something like PROTON_DXVK_USE_DX11=1 or something like that, but this was a command I found in the forums for the demo, which are gone now.

**Comment by DerSandmann-Badcode** [comment]
It is something they added after the first demo, which is a pain. The original first demo loads 100% fine. There were some graphics issues (looked like z fighting) but it was 100% operational.
![image](https://user-images.githubusercontent.com/8218786/47634773-8adc6d80-dbb7-11e8-95aa-3bd5fb81cf53.png)

**Comment by Sbte** [comment]
@Inotamira In your log it says at the bottom:

```
System information:
    Wine build: wine-3.17 (Staging)
```

The latest version of wine that proton uses is 3.16 without staging. Did you compile proton by yourself with a different version of wine or something?

@DerSandmann-Badcode Do you know where I could find this demo?

**Comment by ghost** [comment]
I'm using the beta versions of proton within steam itself as I'm a steam beta tester. I have the old demo, if you have some way for me to get it to you I can.

**Comment by DerSandmann-Badcode** [comment]
> @Inotamira In your log it says at the bottom:
> 
> ```
> System information:
>     Wine build: wine-3.17 (Staging)
> ```
> The latest version of wine that proton uses is 3.16 without staging. Did you compile proton by yourself with a different version of wine or something?
> 
> @DerSandmann-Badcode Do you know where I could find this demo?

I had it from their first demo (Six months ago now?) - I saved the files when I heard wind of them cancelling the release. I'm curious what has changed since then and now - I'll try digging a little deeper.

**Comment by ghost** [comment]
> > @Inotamira In your log it says at the bottom:
> > ```
> > System information:
> >     Wine build: wine-3.17 (Staging)
> > ```
> > The latest version of wine that proton uses is 3.16 without staging. Did you compile proton by yourself with a different version of wine or something?
> > @DerSandmann-Badcode Do you know where I could find this demo?
> 
> I had it from their first demo (Six months ago now?) - I saved the files when I heard wind of them cancelling the release. I'm curious what has changed since then and now - I'll try digging a little deeper.

I think they completely swapped out the rendering methods from what I can tell, they initially used something more open source. Though I'm not sure what specifically, mfplat wasn't used at all, but that had a few stability issues, and it looks like the whole effort was simply to switch it over to simply using mfplat for all the media based functions.

**Comment by Ahmed-E-86** [comment]
**A possible fix for the game:**
https://github.com/doitsujin/dxvk/issues/728#issuecomment-459839962

**Comment by Sbte** [comment]
I already fixed all mfplat related crashes. The issue now is that it only shows a black screen, which is most likely d3d11 related. It works with neither wine, wine-staging or dxvk.

**Comment by ahmed-elsayed2017** [comment]
> I already fixed all mfplat related crashes. The issue now is that it only shows a black screen, which is most likely d3d11 related. It works with neither wine, wine-staging or dxvk.

That will be helpful for the people looking for a fix for the game.

**Comment by Sbte** [comment]
They now also added a DirectShow backend for the movies, so mfplat is no longer required. See

https://steamcommunity.com/app/803600/discussions/0/1776010325109084908/

The screen is still black after the intro movie though.

**Comment by Dagmatto** [comment]
As of the time of writing, installing d3dcompiler_47 through the latest winetricks makes the game work. Damn near flawless too, but the intro video doesn't play.

**Comment by Sbte** [comment]
Thanks for the information. You can fix the videos by setting

```
MovieType=1
```

in

```
c:/users/[name]/Application Data/Nippon Ichi Software, Inc/Disgaea 5 Complete/[#]/config.ini
```

as per 

https://steamcommunity.com/app/803600/discussions/0/1776010325109084908/

Edit: I just checked and the videos do seem to work in wine, but not in Proton.

**Comment by DuloT34** [comment]
@Sbte is there any workarounds for now to make the videos work?

**Comment by Sbte** [comment]
Not that I know of. It is weird that it works in wine with DirectShow, but not in Proton though. However, Nikolay Sivov is working on implementing MF in wine at the moment, so it might be that in future versions of Proton, videos just start working with the MF backend.

There's only one movie in the Data folder by the way, which is the intro movie, so if you want to watch it, you can just watch it manually as well. I don't think movies affect gameplay at all.

**Comment by DuloT34** [comment]
@Sbte Oh, thank you for the info. I was thinking there are a lot more movies and I will miss out on the plot until this problem is fixed.

**Comment by DerSandmann-Badcode** [comment]
works pretty flawlessly in the latest proton

**Comment by scottAnselmo** [comment]
Doesn't launch for me w/ latest Proton on latest version of Manjaro. I've noticed for the past couple of the months some version of the Steam client or Manjaro seems to have broken launchability of games. Things that worked flawlessly circa Proton 3.7 like YNAB4, or at least launched like Disgaea 5 no longer actually launch. Have tried steam-native, steam-manjaro, and Flatpak Steam.

**Comment by DerSandmann-Badcode** [comment]
For me, Disgaea wouldn't work in Proton 3.7, once I upgraded to 4.29 its been running flawlessly. No sound issues, no graphics issues, no odd crashes. I just happened to pick it up on the recent sale and it has surprised me how well it runs. I think the steam install skipped the start movie (Or set it manually in the ini), but the rest of the ingame scenarios show fine.

Ubuntu 18.04
Nvidia 415.27

![image](https://user-images.githubusercontent.com/8218786/60870132-5a8bb980-a284-11e9-8f5f-f9178bb49f3a.png)

**Comment by DanMan** [comment]
I get choppy, stuttering music with Proton 4.11-2 on Ubuntu 18.04.02. It also froze when switching from windowed to fullscreen. But it kept the setting and it works fine after re-launching the game.

I eventually fixed the stuttering by setting `default-fragment-size-msec = 20` in  the Pulseaudio `daemon.conf` instead of the default `25`. You might have to go even lower.

https://github.com/ValveSoftware/SteamOS/issues/433

**Comment by CarnageDevs** [comment]
Has anyone managed to get videos working in the game? I've tried a number of things (Proton GE with config.ini set to 2 for Media Foundation, Proton 5.0-10 with `directshow` and config.ini set to 1, disabling `winegstreamer`, installing LAV Filters etc.) and nothing seems to have worked so far.

The game runs fine otherwise on Proton however, though with Proton-GE I needed to disable `winegstreamer` in winecfg or the game audio would crackle severely at the start. In addition, using the `PROTON_USE_WINED3D=1` argument seems to cause the game to crash as soon as it reaches the menu.

Would love to get this figured out if possible since the game seems to use both DirectShow and Media Foundation (depending on what the MovieType in config.ini is set to) so I thought getting videos to work wouldn't be too difficult, but even with Proton GE which has MF it doesn't load the video, and Proton Experimental doesn't even show color bars (because work on MF is being done) in place of the video.

**Comment by DerSandmann-Badcode** [comment]
@Carnage555 What exactly do you mean by videos working? I'm not using any customisations or whatever, it just works out of the box.

Are you talking about the opening movie?

**Comment by CarnageDevs** [comment]
@DerSandmann-Badcode I am indeed referring to the opening movie, everything else works fine but ideally I'd like to have the opening working as well.

**Comment by DerSandmann-Badcode** [comment]
@Carnage555 I don't think WMP (or DirectShow) is fully supported. You'll see similar issues logged with VNs as well. Here's a link to somewhat of the source issue. Not something easily fixed across the board. https://github.com/ValveSoftware/Proton/issues/1464

Looks like this might be something that is being worked on in Wine 6.0. https://www.winehq.org/announce/6.0

**Comment by CarnageDevs** [comment]
@DerSandmann-Badcode I know work on media codecs is being done, but even with VNs I've yet to find one that I can't make work with the aid of protontricks or the GE fork. I figure I just haven't experimented enough, someone managed to get Catherine Classic (a game that's thought to have been waiting for video codec work before it could function) working with just a few tweaks, you can see the solution in the issue tracker for the game.

**Comment by kisak-valve** [comment]
Experimental proton, amd radeon Archlinux Disgaea 5 not working

Issue transferred from https://github.com/ValveSoftware/Proton/issues/5767.
@chriztheanvill posted on 2022-04-14T21:23:14:

# Compatibility Report
- Name of the game with compatibility issues: Disgaea 5 complete
- Steam AppID of the game: 803600

## System Information
- GPU: Advanced Micro Devices, Inc. [AMD/ATI] Wani [Radeon R5/R6/R7 Graphics] (rev cc)
- Driver/LLVM version: Amd amd Carrizo (LLVM 13.0.1 DRM 3.44)
- Kernel version:  5.17.2-arch3-1
- Link to full system information report as [Gist](https://gist.github.com/):
- Proton version: Experimental

## I confirm:
- [ ] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.


[steam_log.txt](https://github.com/ValveSoftware/Proton/files/8492253/steam_log.txt)

## Symptoms 
I have this game installed in a laptop with Intel and Nvidia, with Fedora 35 and works fine with the experimental Proton. In the another laptop, with amd and radeon with Archlinux, does not work the game.
Edit: Works on proton 7.0.1

Can be opened, shows 3 screens, and quit the game, before the intro video starts

## Reproduction
Open game
Wait to continue the game.
